object/task.go

```diff
@@ -64,3 +64,37 @@ func (t *Task) Cancel(ctx context.Context) error {
 
 	return err
 }
+
+// SetTaskState sets task state and optionally sets results or fault, as appropriate for state.
+func (t *Task) SetTaskState(ctx context.Context, state types.TaskInfoState, result types.AnyType, fault *types.LocalizedMethodFault) error {
+	req := types.SetTaskState{
+		This:   t.Reference(),
+		State:  state,
+		Result: result,
+		Fault:  fault,
+	}
+	_, err := methods.SetTaskState(ctx, t.Common.Client(), &req)
+	return err
+}
+
+// SetTaskDescription updates task description to describe the current phase of the task.
+func (t *Task) SetTaskDescription(ctx context.Context, description types.LocalizableMessage) error {
```

Comment:
Since this is a Task method, let's remove 'Task' from the name: `SetDescription` instead of `SetTaskDescription`